### ISSUE TITLE
Error: Helper tool

### ISSUE THREAD

**Issue by petermeier1a** [IngmarStein/Monolingual#76]
Unfortunately Monolingual is not working anymore..

OS X: 10.11.4 (El Capitan)
Monolingual: 1.7.3.

The program tries to install a helper tool but after type in the password for OS X, the installation fails...
"Monolingual is trying to install a new helper tool. Type your password to allow this."

**Comment by IngmarStein** [comment]
I'm sorry you run into this installation issue. It's a known bug in OS X 10.11 and I haven't found a workaround besides rebooting your Mac. I've received the following update from Apple Developer Relations yesterday:
> The original report on this issue, Bug ID 19601397, is still under investigation.

**Comment by hatoncat** [comment]
I've rebooted several times and the issue persists. Is there an OpenRadar copy of the bug report?

I noticed Apple closed a related bug as a duplicate of yours, so they at least seem to be aware of it... but rebooting does nothing to remedy the issue for me.

**Comment by IngmarStein** [comment]
They're definitely aware of it. I've spent one of my DTS support incidents on this and they've been trying to come up with workarounds but it's not been successful thus far.

Here's the link to OpenRadar: https://openradar.appspot.com/23143866

I lost hope this will fixed in 10.11. Apple will probably remove this problematic API in El Capitan's successor and will hopefully provide a replacement (but maybe not).

**Comment by gratvador** [comment]
Hi,
what sound strange for me is the fact that on some computers it works every time, or one time, or never… Unfortunately the upgrade to 10.11.5 on my main computer does not help…
As it worked one time on this last, is there a way to return to this "one shot" status by deleting something somewhere?
Regards.

**Comment by IngmarStein** [comment]
Have you tried the commands mentioned in #45?

My hope is on OS X 10.12…

**Comment by gratvador** [comment]
Hello,

tested with or without reboot and now I get this at the first launch :



If I retry to run Monolingual I go into the buckle asking password for the Eternity

**Comment by The-Wolf312** [comment]
Sadly your hopes are dashed, the app still has the issue in OS X 10.12.

**Comment by gratvador** [comment]
Sniff…

Thx

**Comment by The-Wolf312** [comment]
Also you might want to do what you can to support the new APFS, I'm not sure when it's going to replace HFS+ but in the update notes it warned developers that they should ensure their apps support it.

**Comment by IngmarStein** [comment]
I've updated my bug report to indicate that the bug is indeed still present in Sierra (reproducible with Apple's own sample code).

As far as APFS is concerned, I don't think there is anything needed on Monolingual's side. It runs fine on case-sensitive filesystems and uses only high-level APIs for FS operations.

**Comment by cfabianski** [comment]
I got it working following the thread here https://github.com/IngmarStein/Monolingual/issues/48

Just run the following in the terminal and you should be good to go
`sudo launchctl load -w /Library/LaunchDaemons/com.github.IngmarStein.Monolingual.Helper.plist`

**Comment by cfabianski** [comment]
Alternatively, you can use something like this

```bash
sudo find /Applications ! ( -name English.lproj -or -name French.lproj -or -name fr.lproj -or -name en.lproj ) -and -name "*.lproj" -type d -prune -exec rm -rf {} +
```

or

```bash
sudo find /Applications ! \( -name English.lproj -or -name French.lproj -or -name fr.lproj -or -name en.lproj \) -and -name "*.lproj" -type d -prune -exec rm -rf {} +
```

if you are using zsh.

**Comment by hatoncat** [comment]
Yep! launchctl load did the trick.

Would be great to do programmatically - or at least update the helper error string instructing to run the launchctl manually.

**Comment by The-Wolf312** [comment]
You could write a simple script to do it in the mean time.

**Comment by hatoncat** [comment]
I don't even think a script is needed - I'm just suggesting the string update as a simple workaround - so people who don't monitor GitHub can do it. It appears Apple fixed this launchd issue in newer macOS installs - so the more I think about it, I think it wouldn't be worth the effort to fix programmatically.

Really this only impacts people that haven't done a clean install of macOS in years - which I'll admit I'm a part of, but mostly because my system is clean enough to not need it (other than this issue, which I worked around via Target Disk Mode for awhile now anyways).

**Comment by cfabianski** [comment]
> Really this only impacts people that haven't done a clean install of macOS in years

I don’t think that’s the issue. This laptop is from 2016 and I’ve got another one from 2015. On the 2015 it was working perfectly fine while on the 2016 it wasn’t.

**Comment by The-Wolf312** [comment]
I had a laptop from 2014 and it hadn't been working with it for years, now on my new laptop (2017) the app won't run unless I do that command. I have done a clean install within the last few months.

**Comment by The-Wolf312** [comment]
Actually, this gives me an idea for a potential workaround. One could edit the UI on the helper tool error and put in a button that says "Try Again" using this button they could have the application execute the `sudo launchctl load -w /Library/LaunchDaemons/com.github.IngmarStein.Monolingual.Helper.plist` command then ask for the user's password and restart the app.

**Comment by hatoncat** [comment]
Yeah, that's basically what I was thinking in terms of a programatic solution, but I would label the button as "attempt fix" so that people didn't fear Try Again would be repeating the same solution.

That said, at this point most users don't seem to be having this problem. I'd say "Error loading helper tool, try running this command in Terminal: sudo launchctl load -w /Library/LaunchDaemons/com.github.IngmarStein.Monolingual.Helper.plist"

Would be a fine fix.
